openprovider/openprovider.php

```diff
@@ -1312,14 +1314,48 @@ private function manageNameservers(
         $op_domain_request = $api->call('searchDomainRequest', $args);
         $this->logRequest($api);
 
-        if ($op_domain_request->getCode() != 0 || count($op_domain_request->getData()['results']) < 1) {
-            $this->Input->setErrors([
-                'errors' => [
-                    Language::_('OpenProvider.!error.domain.not_exist', true)
-                ]
-            ]);
+        $domain_request_failed = $op_domain_request->getCode() != 0;
 
-            return false;
+        if ($domain_request_failed) {
+            if ($page_on_client_side) {
+                $this->Input->setErrors([
+                    'errors' => [
+                        $op_domain_request->getMessage()
+                    ]
+                ]);
+
+                return false;
+            }
+
+            $vars->error = $op_domain_request->getMessage();
```

Comment:
please move to the separate method and reuse it when needed
I mean this code
$vars->error = ...
this->view->set('vars', $vars);
return $this->view->fetch();